catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ScoreCommand.java

```diff
@@ -0,0 +1,253 @@
+import static org.apache.karaf.shell.support.ansi.SimpleAnsi.COLOR_RED;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Completion;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.support.completers.FileCompleter;
+import org.codice.ddf.commands.util.RuleSet;
+import org.codice.ddf.commands.util.ValidatorConfigReader;
+import org.codice.ddf.commands.util.ValidatorRuleSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.util.Describable;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ValidationException;
+
+@Command(scope = CatalogCommands.NAMESPACE, name = "score", description = "Score an XML file according to a pre-defined rubric.")
+@Service
+public class ScoreCommand implements Action {
+
+    @Argument(name = "fileName", description = "The path to the file that you want to validate", required = true)
+    @Completion(FileCompleter.class)
+    String filename;
+
+    @Reference
+    List<MetacardValidator> validators;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScoreCommand.class);
+
+    RuleSet ruleSet;
+
+    private String ruleSetFilename = Paths.get(System.getProperty("ddf.home"), "etc", "validation-ruleset.xml").toString();
+
+    // Using system out for println is caught by our githooks to prevent bad practice.
+    // We actually need to use it here.
+    protected PrintStream console = System.out;
+
+    private static final String ERROR_COLOR = COLOR_RED;
+
+    // Put all the formatting in one place so it's consistent and easy to change
+    private static final int COLUMN_1_WIDTH = 20;
+    private static final int COLUMN_2_WIDTH = 47;
+    private static final int COLUMN_3_WIDTH = 5;
+    private static final String TITLE = StringUtils.center("Metadata Quality Report", 80);
+    private static final String SEPARATOR = StringUtils.repeat("-", COLUMN_1_WIDTH) + "-+-" + StringUtils.repeat("-", COLUMN_2_WIDTH) + "-+-" + StringUtils.repeat("-", COLUMN_3_WIDTH);
+    private static final String TOTAL = StringUtils.repeat(" ", COLUMN_1_WIDTH + COLUMN_2_WIDTH + COLUMN_3_WIDTH - 7) + "Total  ";
+    private static final String FORMAT = "%-" + COLUMN_1_WIDTH + "s | %-" + COLUMN_2_WIDTH + "s | %" + COLUMN_3_WIDTH + "s %n";
+
+
+    @Override
+    public Object execute() throws Exception {
+
+        try {
```

Comment:
Any reason why you did not use try-with-resources here?

http://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html